Port Most of the Query Support

masojus opened the pull request:
This all seems to build, but I haven't tried it on MacOS or Linux.

masojus commented:
This ports a bunch of Query functionality and tests. We need to exercise it manually very extensively.

masojus commented:
BTW this brings in a ton of HTTP-related test helpers. I'm hoping we can expand on those.

Pothulapati commented:
Looks good to me. I also tried some queries using a `netcore2.0 `Console Application. 

Sorry for making you work on this. Mine didn't work because `Queries` wasn't initialised. :\

Pothulapati commented:
Awesome! Looks like all the Funnel, Multi Analysis are also Ported. What are all left to port now?

baumatron commented on Keen.NetStandard.Test/QueryTest.cs:
```diff
@@ -0,0 +1,1189 @@
+using System.Threading.Tasks;
+
+
+namespace Keen.Net.Test
```
This should be changed to `Keen.NetStandard.Test`

baumatron commented:
A couple of tests are failing on MacOS, I can investigate what's going on there.
![image](https://user-images.githubusercontent.com/8376410/28844051-fb61f21e-76c8-11e7-9b0c-ba504a5f07b4.png)

baumatron commented:
Looks like the serialized JSON only has LF while the expected JSON has CRLF.

masojus commented:
Did you run this on Linux too or just MacOS? I'd expect it to fail there as well. To start with we should be expecting something like `Environment.NewLine` in those tests, but in some of my older notes I jotted down a TODO to trim all that JSON based on a debug flag.

masojus commented:
Alright, I rebased and squashed a few things, made a couple fixes, and we've got 117 passing tests on MacOS and Linux, and 316 passing tests across the net35, net45 and netstandard2.0 binaries on Windows. Assuming AppVeyor comes back clean I'll probably move forward with merging this.
